scripts/irods/test/test_resource_types.py

```diff
@@ -2385,6 +2385,153 @@ def test_auto_repl_on__2941(self):
+            self.admin.assert_icommand("iadmin modrepl logical_path " + logical_path + " replica_number 1 DATA_REPL_STATUS 1")
+
+            # get the stale replica
+            get_filename = "getfile.txt"
+            self.admin.assert_icommand("iget -n 0 " + filename + " " +  get_filename)
+
+            # assert on the stale replica file size
+            self.assertEqual(os.path.getsize(get_filename), 20)
+
+            # get the file without selecting the replica
+            self.admin.assert_icommand("iget -f " + filename + " " +  get_filename)
+
+            # assert on the good replica file size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+
+        finally:
+            # cleanup
+            self.admin.assert_icommand("irm -f " + filename)
+            if os.path.exists(filename):
+                os.remove(filename)
+            if os.path.exists(stale_replica_filename):
+                os.remove(stale_replica_filename)
+            if os.path.exists(get_filename):
+                os.remove(get_filename)
+
+    def test_direct_request_for_cache_archive_replica(self):
+
+        try:
+            filename = "directrequestcachearchivereplica.txt"
+            get_filename = "getfile.txt"
+            lib.make_file(filename, 10, 'arbitrary')
+
+            logical_path = os.path.join(self.admin.session_collection, filename)
+
+            self.admin.assert_icommand("iput " + filename)
+
+            # attempt to get the replica directly from archive - should fail
+            self.admin.assert_icommand(['iget', '-n1', filename, get_filename], 'STDERR', 'DIRECT_CHILD_ACCESS')
+
+            # attempt to get the replica directly from cache - should pass
+            self.admin.assert_icommand(['iget', '-n0', filename, get_filename])
+
+            # assert that the retrieved file is the right size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+            os.remove(get_filename)
+
+            # get the file without requesting replica or resource
+            self.admin.assert_icommand(['iget', filename, get_filename])
+
+            # assert that the retrieved file is the right size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+            os.remove(get_filename)
+
+            # get the file specifying the compound resource - should pass
+            self.admin.assert_icommand(['iget', '-R', 'demoResc', filename, get_filename])
+
+            # assert that the retrieved file is the right size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+            os.remove(get_filename)
+
+        finally:
+            # cleanup
+            self.admin.assert_icommand("irm -f " + filename)
+            if os.path.exists(filename):
+                os.remove(filename)
+            if os.path.exists(get_filename):
+                os.remove(get_filename)
+
+    def test_direct_request_for_cache_archive_replica_prefer_archive_policy(self):
```

Comment:
The test name isn't clear to me. Can we improve it in some way?

Can you explain what this test is about?

Reply:
This is similar to the `test_direct_request_for_cache_archive__issue_6926` but with the policy set to prefer archive (different code path).

What about this name?  `test_direct_request_for_cache_archive_replica_policy_set_to_prefer_archive__issue_6926`.

Reply:
After discussion, this has been updated to `test_direct_request_for_cache_archive_replicas_when_policy_is_set_to_prefer_archive__issue_6926`.